### DISCUSSION TITLE
Plotting: customizable legend in config

### DISCUSSION THREAD

**Discussion by JulesVandenbroeck** [columnflow/columnflow#362] [Ideas]
Hi everyone,

While messing around with plotting in columnflow (cf.PlotVariable1D) I did not find a straightforward way to customize the legend. Is there a possibility to write options for the legend in `config.default_general_settings`? In either case it would be helpful to have ability to add `config.x.default_legend_cfg` when creating the config that gives customization to the style of the legend.
My suggestion would be to add a check if `config.x.default_legend_cfg` exists in `columnflow.plotting.plot_util l166` and if so use this instead:
l166: `"legend_cfg": getattr(config_inst.x,"default_legend_cfg",{}),`

I apologize in case there is already a way to customize the legend and I just did not find it.

Greetings,
Jules

**Comment by mafrahm** [comment]
Thank you for opening the discussion! Indeed, one could overwrite it in the `plot_util`, but I think, a better approach would be to directly update the `style_config` that is handed to the plotting function from the task. I opened a PR (#364), where I propose a possible implementation.

**Reply by JulesVandenbroeck** [reply]
Thank you for the answer, this is indeed a more elegant approach. I will close this discussion and follow the PR ([#364](https://github.com/columnflow/columnflow/pull/364))!
